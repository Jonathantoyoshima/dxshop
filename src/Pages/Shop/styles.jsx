import styled from 'styled-components';

export const List = styled.div`
	box-sizing: border-box;
	border: 1px solid
		${(props) => props.theme.color.basic.white};
	margin: auto;
	padding: 20px;
    display: grid;
    grid-gap: 1rem;
    grid-template-columns: repeat(3, minmax(200px, 1fr));
`;

export const Title = styled.h1`
    font-size: 20px;
    text-transform: uppercase;
`;
