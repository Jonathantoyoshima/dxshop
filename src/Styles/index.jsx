import styled from 'styled-components';

export const theme = {
	typography: {
		normal: `font-family: Arial, sans; letter-spacing: 0.1rem;`,
	},
	color: {
		basic: {
			black: '#0c0c0c',
			white: '#fff',
		},
		primary: {
			main: '#4f6f8c',
			light: '#b4c4d9',
			dark: '#254559',
		},
		secondary: {
			light: '#4D4D99',
			dark: '#011c26',
		},
	},
};

export const General = styled.div`
	${(props) => props.theme.typography.normal};
`;

export const Container = styled.div`
	position: relative;
	width: 100%;
	max-width: 1080px;
	padding: 0 16px;
`;

export const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	color: ${(props) => props.theme.color.basic.white};
	background-color: ${(props) =>
		props.theme.color.secondary.dark};
`;

export const Button = styled.button`
	padding: 10px 30px;
	text-transform: uppercase;
	letter-spacing: 0.3rem;
	border-radius: 10px;
	background-color: ${(props) =>
		props.theme.color.primary.dark};
	border: 1px solid
		${(props) => props.theme.color.basic.white};
	color: ${(props) => props.theme.color.basic.white};
	cursor: pointer;
	transition: all 1s ease;
	:hover {
		background-color: ${(props) =>
			props.theme.color.primary.light};
		border: 1px solid
			${(props) => props.theme.color.primary.dark};
		color: ${(props) => props.theme.color.primary.dark};
	}
`;

export const Input = styled.input`
	padding: 8px;
	width: 100%;
	box-shadow: inset -5px -5px 20px 0px rgba(0, 0, 0, 0.5);
`;
