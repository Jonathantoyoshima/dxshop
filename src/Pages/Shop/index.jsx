import React from 'react';
import { Button, Container, Wrapper } from 'Styles';
import { List, Pages, Title } from './styles';
import Card from 'Components/Card';
import Header from 'Components/Header';
import SearchBar from 'Components/SearchBar';
import CartButton from 'Components/CartButton';
import Cart from 'Images/icon/cart.svg';
import { StoreContext } from 'Store';

const Shop = () => {
  const [store] = React.useContext(StoreContext);
  const [page, setPage] = React.useState(0);
  let ShowData = store.data
    .filter((item) => !store.cart.some((el) => el.id === item.id))
    .filter((item) =>
      Number(store.chooseCategory) >= 0 ? Number(item.idCategory) === Number(store.chooseCategory) : true
    );
  let numberOfPages = Math.ceil(ShowData.length / 3);
  let initial = page * 3;
  let end = page * 3 + 3;

  return (
    <Wrapper>
      <Header>
        <Container>Logo</Container>
        <CartButton src={Cart} to={'/checkout'} alt='cart' />
      </Header>
      <Container>
        <SearchBar onChange={() => setPage(0)} />
        <Title>Lista de Produtos:</Title>
        <List>
          {ShowData.slice(initial, end).map((item, index) => (
            <Card key={index} title={item.name} id={item.id} />
          ))}
        </List>
        <Pages hide={numberOfPages === 0}>
          <Button disabled={page <= 0} onClick={() => setPage(page - 1)}>
            {'<<<'}
          </Button>
          {`PÃ¡gina ${page + 1}/${numberOfPages}`}
          <Button disabled={page + 1 >= numberOfPages} onClick={() => setPage(page + 1)}>
            {'>>>'}
          </Button>
        </Pages>
      </Container>
    </Wrapper>
  );
};

export default Shop;
