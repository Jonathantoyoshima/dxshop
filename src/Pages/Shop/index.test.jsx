import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from 'Styles';
import Shop from './index';

test('Render Build Shop Snapshot', () => {
  const component = shallow(
    <ThemeProvider theme={theme}>
      <Shop />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot();
});
