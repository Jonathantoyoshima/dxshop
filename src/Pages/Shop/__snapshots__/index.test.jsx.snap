// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build Shop Snapshot 1`] = `ShallowWrapper {}`;
